subsys/net/l2/canbus/canbus.c

```diff
@@ -1,49 +1,1211 @@
 /*
- * Copyright (c) 2019 Intel Corporation.
+ * Copyright (c) 2019 Alexander Wachter.
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #include <logging/log.h>
-LOG_MODULE_REGISTER(net_l2_canbus, LOG_LEVEL_NONE);
+LOG_MODULE_REGISTER(net_l2_canbus, CONFIG_NET_L2_CANBUS_LOG_LEVEL);
 
 #include <net/net_core.h>
 #include <net/net_l2.h>
 #include <net/net_if.h>
 #include <net/net_pkt.h>
-#include <net/socket_can.h>
+#include <net/can.h>
+#include "canbus_internal.h"
+#include <6lo.h>
+#include <timeout_q.h>
+#include <string.h>
+#include <misc/byteorder.h>
 
-static inline enum net_verdict canbus_recv(struct net_if *iface,
-					   struct net_pkt *pkt)
+#define NET_CAN_WFTMAX 2
+#define NET_CAN_ALLOC_TIMEOUT K_MSEC(100)
+
+/*Minimal separation time betwee frames*/
+#define NET_CAN_STMIN 0
+
+#define NET_CAN_DAD_SEND_RETRY 5
+#define NET_CAN_DAD_TIMEOUT K_MSEC(100)
+
+static struct can_net_isotp_tx_ctx tx_ctx[CONFIG_NET_PKT_TX_COUNT];
```

Comment:
Could you create one context struct and then place these local variables there? We typically do it like that in the other L2 drivers as it is easier to read and review the code